### PR TITLE
Release/v0.32.1

### PR DESCRIPTION
**Pull request by marbar3778**
* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+  mempool indicate to the ABCI app whether a CheckTx request is a recheck or
+  not. see [docs](https://github.com/tendermint/tendermint/blob/eddb433d7c082efbeaf8974413a36641519ee895/docs/spec/abci/apps.md#mempool-connection)
+
+### IMPROVEMENTS:
+
+- [rpc] [\#3700](https://github.com/tendermint/tendermint/issues/3700) Make possible to set absolute paths for TLS cert and key (@climber73)
+
+### BUG FIXES:
+
+- [p2p] [\#3338](https://github.com/tendermint/tendermint/issues/3338) Prevent "sent next PEX request too soon" errors by not calling
```
Comment (melekes) [inline CHANGELOG.md]: this should probably be back ported

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,52 @@
 # Changelog
 
+## v0.32.1
+
+*July 12, 2019*
+
+Special thanks to external contributors on this release: 
+@ParthDesai, @climber73, @jim380, @ashleyvega
+
+This release contains breaking changes to our libs folder and ABCI namely:
```
Comment (ebuchman) [inline CHANGELOG.md]:
```suggestion
This release contains a minor enhancement to the ABCI and some breaking changes to our libs folder, namely:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
+- CheckTx requests include a `CheckTxType` enum that can be set to `Recheck` to indicate to the application that this transaction was already checked/validated and certain expensive operations (like checking signatures) can be skipped
+- Removed various functions from `libs` pkgs
+  `GoPath()`, `Prompt()` and `IsDirEmpty()` functions from `os.go`
```
Comment (ebuchman) [inline CHANGELOG.md]:
```suggestion
```

No need to provide these details here if they're present below.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,52 @@
...
+  - [libs] [\#2432](https://github.com/tendermint/tendermint/issues/2432) Remove unused `common/heap.go` file (@marbar3778)
+  - [libs] Remove unused `date.go`, `io.go`. Remove `GoPath()`, `Prompt()` and `IsDirEmpty()` functions from `os.go` (@marbar3778)
+  - [libs] Remove unused `FailRand()` func and minor clean up to `fail.go`(@marbar3778)
+
+### FEATURES:
+
+- [node] Add variadic argument to `NewNode` to support functional options, allowing the Node to be more easily customized. 
+- [node][\#3730](https://github.com/tendermint/tendermint/pull/3730) Add `CustomReactors` option to `NewNode` allowing caller to pass
+  custom reactors to run inside Tendermint node (@ParthDesai)
+- [abci] [\#2127](https://github.com/tendermint/tendermint/issues/2127) ABCI / mempool: Add a "Recheck Tx" indicator. Breaks the ABCI
```
Comment (ebuchman) [inline CHANGELOG.md]: What happened here?

The breaking change to the Go API should be listed under that section for breaking changes. The non breaking change to the ABCI itself should be listed here as a feature.

Comment (ebuchman) [reply CHANGELOG.md]: We should also be specific about what got added here, namely the CheckTxType

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,52 @@
 # Changelog
 
+## v0.32.1
+
+*July 12, 2019*
```
Comment (ebuchman) [inline CHANGELOG.md]:
```suggestion
*July 15, 2019*
```

:)

**Pull request merged**